src/d3a/models/strategy/settlement/strategy.py

```diff
@@ -0,0 +1,199 @@
+                initial_rate=initial_selling_rate,
+                final_rate=final_selling_rate,
+                fit_to_limit=True,
+                energy_rate_change_per_update=None,
+                update_interval=duration(minutes=self._update_interval),
+                rate_limit_object=max)
+
+    def event_market_cycle(self, strategy: BaseStrategy) -> None:
+        """
+        Should be called by the event_market_cycle of the asset strategy class, posts
+        settlement bids and offers on markets that do not have posted bids and offers yet
+        :param strategy: Strategy object of the asset
+        :return: None
+        """
+        self.bid_updater.update_and_populate_price_settings(strategy.area)
+        self.offer_updater.update_and_populate_price_settings(strategy.area)
+        for market in strategy.area.settlement_markets.values():
+            energy_deviation_kWh = strategy.state.get_unsettled_deviation_kWh(market.time_slot)
+
+            if strategy.state.should_post_bid(market.time_slot):
+                try:
+                    strategy.post_first_bid(
+                        market, energy_deviation_kWh * 1000.0,
+                        self.bid_updater.initial_rate[market.time_slot]
+                    )
+                except MarketException:
+                    pass
+
+            if strategy.state.should_post_offer(market.time_slot):
+                try:
+                    strategy.post_first_offer(
+                        market, energy_deviation_kWh,
+                        self.offer_updater.initial_rate[market.time_slot]
+                    )
+                except MarketException:
+                    pass
+
+    def event_tick(self, strategy: BaseStrategy) -> None:
+        """
+        Update posted settlement bids and offers on market tick
+        :param strategy: Strategy object of the asset
+        :return: None
+        """
+        self.bid_updater.increment_update_counter_all_markets(strategy)
+        self.offer_updater.increment_update_counter_all_markets(strategy)
+        for market in strategy.area.settlement_markets.values():
+            self.bid_updater.update(market, strategy)
+            self.offer_updater.update(market, strategy)
+
+    def _get_settlement_market_by_id(
+            self, strategy: BaseStrategy, market_id: str) -> Optional["Market"]:
+        markets = [m for m in strategy.area.settlement_markets.values() if m.id == market_id]
+        if not markets:
+            return None
+        return markets[0]
+
+    def event_bid_traded(self, strategy: BaseStrategy, market_id: str, bid_trade: Trade) -> None:
+        """
+        Updates the unsettled deviation with the traded energy from the market
+        :param strategy: Strategy object of the asset
+        :param market_id: Id of the market that the trade took place
+        :param bid_trade: Trade object
+        :return: None
+        """
```

Comment:
Nit: We decided to go with the [google style doctring](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)
```suggestion
        """
        Update the unsettled deviation with the traded energy from the market.
        Args:
            strategy: Strategy object of the asset
            market_id: Id of the market that the trade took place
            bid_trade: Trade obje
        Returns: None
        """
```

Reply:
Thank you, my Pycharm messed up the settings again. Will change it to google style